src/libraries/Common/src/Interop/Unix/System.Security.Cryptography.Native/Interop.Crypto.cs

```diff
@@ -71,13 +72,13 @@ internal static int BioTell(SafeBioHandle bio)
             byte usedDefault;
             IntPtr ptr = GetX509RootStorePath_private(&usedDefault);
             defaultPath = (usedDefault != 0);
-            return Marshal.PtrToStringUTF8(ptr);
+            return Utf8StringMarshaller.ConvertToManaged(ptr);
```

Comment:
Also, please change the return type of `GetX509RootStorePath_private` to `byte*`.